### PR TITLE
Display alpha values in Heatmap tooltip

### LINKED ISSUE #876: [Heatmap] Show alpha value in tooltip when visualizing phase+amp of complex datasets

### PR DESCRIPTION
**Pull request by loichuder**
Fix #876

### CODE REVIEW

File: packages/app/src/vis-packs/core/complex/MappedComplexVis.tsx [hunk 1]
```diff
@@ -95,6 +87,15 @@ function MappedComplexVis(props: Props) {
...
         ordinateParams={axisMapping[dimMapping.indexOf('y')]}
+        alpha={
+          visType === ComplexVisType.PhaseAmplitude
+            ? {
+                array: amplitudeArray,
+                domain: amplitudeDomain,
+                label: 'Amp.',
+              }
+            : undefined
+        }
```
Comment (loichuder) [inline packages/app/src/vis-packs/core/complex/MappedComplexVis.tsx:98]: I changed the signature of `Heatmap` to constrain that `alphaArray` and `alphaDomain` must be either both supplied or both `undefined`.

With the previous signature, we could have `alphaArray` defined and `alphaDomain` `undefined` leading to a silent failure: https://h5web-docs.panosc.eu/?path=/story/visualizations-heatmapvis--alpha&args=alphaDomain:!null

Comment (axelboc) [reply packages/app/src/vis-packs/core/complex/MappedComplexVis.tsx:98]: Great idea 💯

File: packages/lib/src/vis/heatmap/HeatmapVis.tsx [hunk 2]
```diff
@@ -111,6 +109,11 @@ function HeatmapVis(props: Props) {
                 <div className={styles.tooltipValue}>
                   <strong>{formatTooltipVal(dataArray.get(yi, xi))}</strong>
                   {dtype && <em>{` (${formatNumType(dtype)})`}</em>}
+                  {alpha && (
+                    <em>{` ${alpha.label ?? 'alpha'}=${formatTooltipVal(
+                      alpha.array.get(yi, xi)
+                    )}`}</em>
+                  )}
```
[the remark this answers is not in the document]
Comment (axelboc) [reply packages/lib/src/vis/heatmap/HeatmapVis.tsx:116]: Yeah, if we had a label for the main value like on the `LineVis` (`ordinateLabel`), we could have two lines like this:

```
<value-label>=-4
<alpha-label>=4
```

... but we don't have a label. The `title` would be too long, so we'd have to add a new prop for it, which is annoying. Not sure it's worth it. Maybe just put the alpha part in parens?

**Review by axelboc — APPROVED** [review]

File: packages/lib/src/vis/heatmap/HeatmapVis.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (loichuder) [reply packages/lib/src/vis/heatmap/HeatmapVis.tsx:116]: Sounds good to me

**Pull request merged**